pandas/core/frame.py

```diff
@@ -6475,123 +6475,122 @@ def append(self, other, ignore_index=False,
     def join(self, other, on=None, how='left', lsuffix='', rsuffix='',
              sort=False):
         """
-        Join columns with other DataFrame either on index or on a key
-        column. Efficiently Join multiple DataFrame objects by index at once by
+        Join columns of another DataFrame.
+
+        Join columns with `other` DataFrame either on index or on a key
+        column. Efficiently join multiple DataFrame objects by index at once by
         passing a list.
 
         Parameters
         ----------
-        other : DataFrame, Series with name field set, or list of DataFrame
+        other : DataFrame, Series, or list of DataFrame
             Index should be similar to one of the columns in this one. If a
             Series is passed, its name attribute must be set, and that will be
-            used as the column name in the resulting joined DataFrame
-        on : name, tuple/list of names, or array-like
+            used as the column name in the resulting joined DataFrame.
+        on : str, list of str, or array-like, optional
```

Comment:
What's the difference between `"default None"` vs. `"optional"` ?

One might argue that any argument with a default argument is optional, but what that default argument is would be nice to know.

Reply:
I don't remember where this comes from, but the idea is that if the `None` is the value being used, then we use `default None`, and if `None` is just a sentinel to say, this does not apply here, then we use `optional`.

Reply:
Hmm...I see.  At the risk of bucking existing trends, I wonder if that level of semantic is clear to end users, or whether we should be explicitly telling people that default arguments are just sentinel values?

Reply:
I find it intuitive enough, and we usually try to add in the description more information, including what to expect when using the default value. But if you think it's worth discussing in more detail, I think the best would be to open an issue. I'm open to change the convention.

Btw, just to give some context. I'm trying to close all the old PRs related to the documentation (this was I reopened, but the content is from a stale PR). And what I mean, is that I'm happy merging the changes as far as the changes are an improvement, and everything is correct. For other improvements I'd prefer to wait for a new PR (otherwise, merging the pending 50 PRs will be my full time job, and I prefer new contributors to do the work on the content, so I can focus on reviewing).

Reply:
> I prefer new contributors to do the work on the content, so I can focus on reviewing).

You and me both 😉 

Wasn't intending to drag this out, but wanted to bring this up in case others had already given thought to this, as I am not as involved in the doc's as you are for example.

Reply:
hehe, sure, the review was great, and I didn't realize about the notes, but that had to be changed. I was just letting you know, as you'll surely be reviewing other PRs I'm fixing myself, so they don't stay in the pile of pending PRs forever.